packages/venia-ui/lib/components/CartPage/ProductListing/productListingFragments.js

```diff
@@ -0,0 +1,29 @@
+import gql from 'graphql-tag';
+
+export const ProductListingFragment = gql`
+    fragment ProductListingFragment on Cart {
+        id
+        items {
+            id
+            product {
```

Comment:
The props requested in items should _probably_ be moved to a `productListingProductFragment` object that is then used here. The idea is that each visual component declares what it needs to render, and the `productListing` component is really just a wrapper that passes the data down to the `product` component anyways.

Reply:
I like this idea, a great extension of what you did with PriceSummary. This did bring to my attention we're inconsistently colocating fragments though, and should probably decide on a single location.

1. Dedicated file (`<componentName>Fragments.js`)
2. Named export in component (eg. `discountSummary`)

Of course the Apollo docs use static properties, which we know has the issue of cyclical dependencies when children need parent fragments (also an issue with named export). Maybe this is another reason to switch to a single root component query, and use apollo-anywhere filter. This still would require `CartPageFragment` be exported outside of the root component though...

Tough questions, but I'm okay iterating on it and not solving in the scope of this PR.

Reply:
Yea I just wanted to call it out :)